src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -593,37 +637,73 @@ private static TextSpan GetSpan(Stack<SyntaxNodeOrToken> stack, int first, int l
             return TextSpan.FromBounds(start, end);
         }
 
-        private static Stack<SyntaxNodeOrToken> Combine(Stack<SyntaxNodeOrToken> first, Stack<SyntaxNodeOrToken> next)
```

Comment:
:memo: This went from amortized O(n) to amortized O(1) in practice (next was typically 1 element, first contained many).